Source/Core/Core/PowerPC/JitCommon/JitCache.cpp

```diff
@@ -128,72 +124,70 @@ int JitBaseBlockCache::AllocateBlock(u32 em_address)
   b.msrBits = MSR & JitBlock::JIT_CACHE_MSR_MASK;
   b.linkData.clear();
   num_blocks++;  // commit the current block
-  return num_blocks - 1;
+  return &b;
 }
 
-void JitBaseBlockCache::FinalizeBlock(int block_num, bool block_link, const u8* code_ptr)
+void JitBaseBlockCache::FinalizeBlock(JitBlock& b, bool block_link, const u8* code_ptr)
 {
-  JitBlock& b = blocks[block_num];
   if (start_block_map.count(b.physicalAddress))
   {
     // We already have a block at this address; invalidate the old block.
     // This should be very rare. This will only happen if the same block
     // is called both with DR/IR enabled or disabled.
     WARN_LOG(DYNA_REC, "Invalidating compiled block at same address %08x", b.physicalAddress);
-    int old_block_num = start_block_map[b.physicalAddress];
-    const JitBlock& old_b = blocks[old_block_num];
+    JitBlock& old_b = *start_block_map[b.physicalAddress];
     block_map.erase(
         std::make_pair(old_b.physicalAddress + 4 * old_b.originalSize - 1, old_b.physicalAddress));
-    DestroyBlock(old_block_num, true);
+    DestroyBlock(old_b, true);
   }
-  start_block_map[b.physicalAddress] = block_num;
+  u32 block_num = &b - &blocks[0];
```

Comment:
`FastLookupEntryForAddress` returns an `int&`, so you'd need to do a cast, otherwise:

```cpp
 FastLookupEntryForAddress(b.effectiveAddress) = block_num;
```

does implicit sign conversion. 

`&b - &blocks[0]` would return a 64-bit signed type (ptrdiff_t), so this would also do implicit truncation, as well as said sign-conversion.

`block_num` can also be const as far as I can tell.